lib/fog/azurerm/models/network/network_security_group.rb

```diff
@@ -48,6 +48,32 @@ def destroy
           service.delete_network_security_group(resource_group, name)
         end
 
+        def update(hash = {})
+          raise('Provided hash can not be empty.') if hash.empty? || hash.nil?
+
+          if !hash[:name].nil? || !hash[:location].nil? || !hash[:resource_group].nil? || !hash[:id].nil?
+            raise('You can not modify id:, name:, location: or resource_group:')
```

Comment:
Update the messaging "You can not update id, name, location or resource_group of network security group"